CanHazFunny/CanHazFunny/Jester.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+
+namespace CanHazFunny
+{
+	public class Jester : IJokeService, IJokeOutput
+	{
+		// redundant _jokeService field, but else leads to infinite regress
+		private IJokeService _jokeService;
+		private IJokeService jokeService { get => _jokeService; set => _jokeService = value ?? throw new ArgumentNullException();  }
```

Comment:
When throwing ArgumentNullExceptions always specify the argument name.
```suggestion
		private IJokeService jokeService { get => _jokeService; set => _jokeService = value ?? throw new ArgumentNullException(nameof(value));  }
```